### ISSUE TITLE
Add option to disallow CONSTANT_CASE in `camelcase` rule

### ISSUE THREAD

**Issue by sindresorhus** [eslint/eslint#10605]
**What rule do you want to change?**

`camelcase`

**Does this change cause the rule to produce more or fewer warnings?**

No change

**How will the change be implemented? (New option, new default behavior, etc.)?**

New option. Could be called `ignoreConstantCase` and be `false` by default.

Ignoring constant-case is too opinionated and should be configurable. I don't want constant-case in my code as it looks ugly and there's no way to enforce when to use it, so it's just used very inconsistently. JS doesn't have real constants anyway, so I prefer keeping it all normal camel-case.

**Please provide some example code that this change will affect:**

```js
const FOO_BAR = true;
```

**What does the rule currently do for this code?**

It allows it.

**What will the rule do after it's changed?**

Disallow it and require the following instead:

```js
const fooBar = true;
```

**Comment by platinumazure** [comment]
Hi @sindresorhus, thanks for the issue.

I think this is a reasonable request and will support it (especially since the current all caps exception can seem arbitrary). However, I do want to point out that this can already be achieved using the [`id-match` rule](https://eslint.org/docs/rules/id-match), if you want to configure this for a current project.

**Comment by sindresorhus** [comment]
@platinumazure So it's final that this will not make it into ESLint? I'm just making sure before I implement this in my own plugin.
